File: HW4/userid/ToyDuck.cpp
#include <iostream>
#include "ToyDuck.h"
#include "FlyNoWay.h"
#include "Quack.h"
#include "LaysToyEggs.h"

ToyDuck::ToyDuck( ) {
    flyBehavior = new FlyNoWay( );
    quackBehavior = new Quack( );
    eggBehavior = new LaysToyEggs( );
}

void ToyDuck::display( ) {
    std::cout << "I'm a toy duck." << std::endl;
}

File: HW14/huan1588/ToyDuck.java
public class ToyDuck extends Duck {
    public ToyDuck( ) {
        quackBehavior = new Quack( );
        flyBehavior = new FlyNoWay( );
        eggBehavior = new LaysToyEggs( );
    }

    public void display( ) {
        System.out.println("I'm a toy duck");
    }
}
